### PR TITLE
Shield Overhaul

### PR DESCRIPTION
**Pull request by Crystalwarrior**
changes:
- bugfix: "Fixed inactive shields (teleshields, energy shields) still blocking projectiles even when inactive"
- tweak: "Shields now have varying protect chances on various damage types"
- tweak: " + Changeling Shield:   melee = 60, bullet = 50, laser = 40, energy = 30"
- tweak: " + Tray Shield:         melee = 60, bullet = 50, laser = 40, energy = 30 (Now trayshields break based on the amount of taken damage overall, making them less RNG and more reliable)"
- tweak: " + Bo Staff:            melee = 50, bullet = 50, laser = 50, energy = 50"
- tweak: " + Riot Shield:         melee = 70, bullet = 60, laser = 40, energy = 30 (Added slowdown when deployed, can be deployed similar to telescopic/energy shield)"
- tweak: " + Energy Shield:       melee = 60, bullet = 70, laser = 60, energy = 40 (still 100% reflect chance on energy projectiles tho)"
- tweak: " + Telescopic Shield:   melee = 50, bullet = 30, laser = 20, energy = 20"
- tweak: " + Roman Shield:        melee = 60, bullet = 50, laser = 50, energy = 50"

### CODE REVIEW

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -1,11 +1,60 @@
...
+	throw_range = 3
+	w_class = 4
+	materials = list(MAT_GLASS=7500, MAT_METAL=1000)
+	origin_tech = "materials=2"
+	block_chance = list(melee = 70, bullet = 50, laser = 50, energy = 50) //Skeleton-exclusive atm, sooo
+
+//Deployable shields, woo!
+//Basically "defensive stance" for shields like riot, etc.
+/obj/item/weapon/shield/deployable
+	block_chance = list(melee = 0, bullet = 0, laser = 0, energy = 0)
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/shields.dm:26]: Woah, are you telling me an undeployed shield does _nothing_?

Comment (JohnGinnane) [inline code/game/objects/items/weapons/shields.dm:26]: Nevermind...

File: code/game/objects/items/weapons/shields.dm [hunk 2]
```diff
@@ -15,70 +64,70 @@
...
+	if(active && user)
+		attack_self(user)
 
-/obj/item/weapon/shield/energy
+/obj/item/weapon/shield/deployable/energy
 	name = "energy combat shield"
 	desc = "A shield capable of stopping most melee attacks. Protects user from almost all energy projectiles. It can be retracted, expanded, and stored anywhere."
-	icon = 'icons/obj/weapons.dmi'
-	icon_state = "eshield0" // eshield1 for expanded
+	icon_state = "eshield"
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/shields.dm:92]: Should this not be "eshield0" so when it's created it's inactive?

File: code/game/objects/items/weapons/shields.dm [hunk 3]
```diff
@@ -15,70 +64,70 @@
...
+		H << "<span class='warning'>You beat yourself in the head with [src].</span>"
+		H.take_organ_damage(force)
 	active = !active
 	icon_state = "eshield[active]"
 
 	if(active)
-		force = 10
+		block_chance = block_chance_deployed
+		slowdown = slowdown_deployed
+		force = force_deployed
```
Comment (JohnGinnane) [inline code/game/objects/items/weapons/shields.dm:116]: How come force has a force_deployed variable but throwforce, throw_speed, and w_class don't?

**Comment by carlarc** [comment]
How about if a shield gets shot with a laser that deals more than 30 damage (laser cannon, pulse rifle)
It has a chance to melt off? Makes no sense for a peasy riot shield to withstand a pulse.

**Comment by Nokington** [comment]
Not sure about this.

File: code/game/objects/items/weapons/shields.dm (hunk 2 above)
Comment (Crystalwarrior) [inline code/game/objects/items/weapons/shields.dm:92]: Check New() proc for deployable

**Comment by Crystalwarrior** [comment]
@carlarc eehhhhh, I dunno about that tbh.

**Comment by Crystalwarrior** [comment]
@Nokington At the very least it's a great framework for tighter balance. Any suggestions regarding the protection values?

File: code/game/objects/items/weapons/shields.dm (hunk 3 above)
Comment (Crystalwarrior) [inline code/game/objects/items/weapons/shields.dm:116]: Because bleh.

**Comment by blobbernaut** [comment]
why is the tray shield almost as strong as the riot shield? It's a god damn tray

**Comment by Crystalwarrior** [comment]
Because the tray can very easily break and let the attack connect

**Comment by myontasm** [comment]
Iffy on this, I'll think about it.

**Comment by carlarc** [comment]
tray shield sucks too fucking much, remove the breaking chance or make it really small and balance protection around that.
i mean seriously a guy can hug you and your shield breaks instantly

**Comment by Pyko1** [comment]
Tray shield should have somekind of damage cap it can block before breaking.

**Comment by Pyko1** [comment]
ie. can block total of 50 damage, after that every point of damage blocked from an attack has an increasing chance to break it.

**Comment by Crystalwarrior** [comment]
Yeah I'll probably do that

**Comment by ghost** [comment]
Changeling no drop shield having 70% melee block from 50% seems a bit memey.
Riot shields being portable with a 70% melee block is absurd (We had the exact same problem with telescopic shields being used to power game.)

Otherwise, this is a cool system.

**Comment by Crystalwarrior** [comment]
I'll nerf the changeling shield to 60

Also Riot Shields are NOT portable, they can still not be stuffed in backpacks. You just have to plan ahead when you're supposed to put up a shield, so I feel that riot shield should have 70% melee for that purpose (if you use it now you'll be more of a sitting duck than before, y'know)

**Comment by dreamcrab** [comment]
i'd like to see how this works out, i don't mind. i have no idea how this is gonna work out. +1 for now but i don't mind saying i would revert it

**Comment by Sarcalogo** [comment]
Maybe make riot shields have some slow down too.. makes no sense that you can run that fast with a huge piece of metal or whatever sticking out in front of you.

**Comment by Crystalwarrior** [comment]
They do give you slowdown when deployed

**Comment by Crystalwarrior** [comment]
This has two heads approval, merge-ready?

**Comment by Nebyab** [comment]
yeah okay

**Comment by Crystalwarrior** [comment]
Aight can someone merge

**Pull request merged**